ExadelTimeTrackingSystem/ExadelTimeTrackingSystem.WebAPI/Controllers/ProjectsController.cs

```diff
@@ -3,63 +3,81 @@
     using System;
     using System.Collections.Generic;
     using System.Linq;
+    using System.Threading;
     using System.Threading.Tasks;
     using ExadelTimeTrackingSystem.BusinessLogic.DTOs;
+    using ExadelTimeTrackingSystem.BusinessLogic.Helpers;
     using ExadelTimeTrackingSystem.BusinessLogic.Services.Abstract;
     using ExadelTimeTrackingSystem.Data.Validators;
+    using ExadelTimeTrackingSystem.WebAPI.Configuration;
     using Microsoft.AspNetCore.Mvc;
+    using Microsoft.Extensions.Options;
 
     [ApiController]
     [Route("[controller]")]
     public class ProjectsController : ControllerBase
     {
         private readonly IProjectService _service;
+        private readonly int? _cancellationTokenTimeOut;
 
-        public ProjectsController(IProjectService service)
+        public ProjectsController(IProjectService service, IOptionsMonitor<TimeOutSettings> config)
         {
-          _service = service;
+            _service = service;
+            _cancellationTokenTimeOut = config.CurrentValue.TimeOutSeconds;
```

Comment:
lets save options to private field, and in methods call _options.CurrentValue.TimeOutSeconds, to always track last changes, otherwise now you take current value once and then use it all the time, no tracking is performed